rflx/model.py

```diff
@@ -1822,6 +1820,25 @@ def proven(self, skip_proof: bool = False) -> Message:
 
     @ensure(lambda result: valid_message_field_types(result))
     def merged(self) -> "UnprovenMessage":
+        def prune_dangling_states(
+            structure: List[Link], types: Dict[Field, Type]
+        ) -> Tuple[List[Link], Dict[Field, Type]]:
+            dangling = []
+            progress = True
+            while progress:
+                progress = False
+                states = {x for l in structure for x in (l.source, l.target) if x != FINAL}
+                for s in states:
+                    if not [l for l in structure if l.source == s]:
```

Comment:
Why not `if all(l.source != s for l in structure):`? I think this would be easier to understand.

Reply:
You're right, changed in cd468f1.